### PR TITLE
Add caddy configuration

### PR DESCRIPTION
**Pull request by bryan-gilbert**
See comments and provided docs for details

### CODE REVIEW

File: ppr-ui/Caddy/Readme.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Client UI Caddy
```
Comment (jguertin) [inline ppr-ui/Caddy/Readme.md]: A subtle file naming convention, but for consistency.  Our other readme's are all uppercase README.md.

File: ppr-ui/openshift/ppr-ui-bc.yaml [hunk 2]
```diff
@@ -28,6 +33,8 @@ objects:
         paths:
         - destinationDir: ./
           sourcePath: /dist
+        - destinationDir: ./
+          sourcePath: /Caddy
```
Comment (jguertin) [inline ppr-ui/openshift/ppr-ui-bc.yaml:37]: What does this do, and does it conflict with `dist`?

File: ppr-ui/openshift/ppr-ui-dc.yaml [hunk 3]
```diff
@@ -72,6 +72,10 @@ objects:
             terminationMessagePath: /dev/termination-log
             terminationMessagePolicy: File
             volumeMounts:
+              - mountPath: /etc/Caddyfile
+                name: coops-web-caddy-config
+                readOnly: true
+                subPath: Caddyfile
```
Comment (jguertin) [inline ppr-ui/openshift/ppr-ui-dc.yaml]: Doesn't this essentially replace the Caddyfile you copied in with a different version from the config map?  Also, please consider renaming to remove `coops`.

**Review by jguertin — COMMENTED** [review]
No specific changes required, but I have a few questions.

File: ppr-ui/openshift/ppr-ui-bc.yaml (hunk 2 above)
Comment (bryan-gilbert) [reply ppr-ui/openshift/ppr-ui-bc.yaml:37]: Actually it doesn't. The destinationDir declares what is available for the Docker commands above

File: ppr-ui/openshift/ppr-ui-dc.yaml (hunk 3 above)
Comment (bryan-gilbert) [reply ppr-ui/openshift/ppr-ui-dc.yaml]: oh oh .. that was a mistake :( . not meant to be left in the code.

File: ppr-ui/Caddy/Caddyfile [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+log stdout
+
+errors stdout
+
+# Source for the rewrite is from the Vue router docs:
+#https://router.vuejs.org/guide/essentials/history-mode.html#example-server-configurations
+rewrite {
+    regexp .*
+    to {path} /
+}
```
Comment (WalterMoar) [inline ppr-ui/Caddy/Caddyfile]: Can't hurt to put in this newline that GitHub is complaining about.

File: ppr-ui/Caddy/d-c.yml [hunk 5]
```diff
@@ -0,0 +1,10 @@
+version: '3'
+services:
+  caddy:
+    image: "abiosoft/caddy:latest"
```
Comment (WalterMoar) [inline ppr-ui/Caddy/d-c.yml:4]: Do we want latest? Ideally we would pull the image from the cluster, but not sure how to do that. Second best would be to match versions of Caddy - although it's a pretty straightforward web server, it more of just a best practice than any kind of concern.

File: ppr-ui/openshift/ppr-ui-inter-bc.yaml [hunk 6]
```diff
@@ -16,6 +16,8 @@ objects:
         kind: ImageStreamTag
         name: ${APP_NAME}:latest
     source:
+      # 1. Checks out the repo; 2 sets the context to be the UI client code; 3 builds the Vue app.
+      # result is an image containing the client code folder and a fully built Vue distribution (in dist folder)
```
Comment (WalterMoar) [inline ppr-ui/openshift/ppr-ui-inter-bc.yaml:20]: You're probably aware, but these comments get stripped out by OpenShift. Any export that follows will not have the comments.

**Comment by WalterMoar** [comment]
Looks good :) Some day I'll have to try out the docker stuff.

**Review by WalterMoar — APPROVED** [review]

File: ppr-ui/Caddy/Caddyfile (hunk 4 above)
Comment (bryan-gilbert) [reply ppr-ui/Caddy/Caddyfile]: added

File: ppr-ui/Caddy/d-c.yml (hunk 5 above)
Comment (bryan-gilbert) [reply ppr-ui/Caddy/d-c.yml:4]: This docker compose file is for local development so taking latest caddy is sufficient and there is no need to use image: "bcgov-s2i-caddy"

File: ppr-ui/openshift/ppr-ui-inter-bc.yaml (hunk 6 above)
Comment (bryan-gilbert) [reply ppr-ui/openshift/ppr-ui-inter-bc.yaml:20]: Right.  Except that we intend to keep our configuration files as the defacto standard and we will not overwrite with an export from OpenShift. We'll cherry pick stuff that we might need.

**Pull request merged**